samples/wfi32iot/basic-sample/src/iotconnect_app.c

```diff
@@ -275,43 +378,126 @@ bool iotconnect_sample_app(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_
     APP_SWITCH_init();
 
     APP_SENSORS_init();
+    
+    //Time in-between data reportings (ms)
+    int loop_time = 5000;
+    
+    //Keeps track of how long the main loop has been running 
+    //for the sensors that have warm-up times
+    int warming_up = 0;
+    
+    //Will hold cumulative delay time for all attached sensors
+    int delay_total = 0;
 
     printf("Detecting attached Click Board sensors...\r\n");
 
-    VAVPRESS_init();
-    if (VAVPRESS_status == VAVPRESS_OK) {
-        printf("VAV Click detected\r\n");
-    } else {
-        printf("VAV Click not detected\r\n");
+    //Array for telling whether or not each click board is present (1 or 0)
+    //Structure is {AQ7, ALT2, ALT4, PHT, T6713, T9602, TH14, ULP, VAV}
+    int connections[] = {0, 0, 0, 0, 0, 0, 0, 0, 0};
+    
+    //AQ7
+    float aq7_data[] = {0, 0, 0, 0};
```

Comment:
Why are we using these awkward array to obtain data? Did we inherit this mess from Microchip and we are trying to keep it the same? I would expect aq7 to be a struct and be dealt as such in the driver.